test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -295,7 +292,7 @@ def test_nonprint_shell_function(self):
                 continue
             func = '() { a=%s; echo XX${a}YY}; }' % ch
             # Adjustments in bash due to ShellShock malware fix in various OS
-            if self.osv in self.osv1 and self.pver >= self.osv1[self.osv]:
+            if self.is_osv1:
                 os.environ['BASH_FUNC_foo()'] = func
```

Comment:
you have the value (BASH_FUNC_foo()) in self.n. can you not use that directly? Using that will get rid of the conditions.

Reply:
The self.n can be None and can't be used directly when exporting via os.environ.

Reply:
@anamikau , I initialize self.n for 'foo' and this will eliminate the need for the if else clause.